### PR TITLE
Use str_ends_with() and str_starts_with()

### PR DESCRIPTION
**Pull request by alexislefebvre**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | no
| License       | MIT
| Doc PR        | no

Use `str_ends_with()` and `str_starts_with()` instead of `substr()`.

(replaces #44476)

### CODE REVIEW

File: src/Symfony/Component/Mime/Tests/Encoder/Rfc2231EncoderTest.php [hunk 1]
```diff
@@ -102,15 +102,15 @@ public function testEncodingAndDecodingSamples()
         $dir = realpath(__DIR__.'/../Fixtures/samples/charsets');
         $sampleFp = opendir($dir);
         while (false !== $encoding = readdir($sampleFp)) {
-            if ('.' == substr($encoding, 0, 1)) {
```
Comment (alexisLefebvre) [inline src/Symfony/Component/Mime/Tests/Encoder/Rfc2231EncoderTest.php:105]: I wonder why the equality test ignored the type of the variables (no `===`).

**Review by GromNaN — CHANGES_REQUESTED** [review]
All updated packages require `symfony/polyfill-php80`, except `symfony/framework: 4.4`, can you add the dependency please.

**Comment by alexislefebvre** [comment]
Thanks for your review.

I'm sorry but `symfony/framework` [doesn't exist](https://packagist.org/?query=symfony%2Fframework). The `composer.json` of `symfony/framework-bundle` already requires `symfony/polyfill-php80` though: https://github.com/symfony/symfony/blob/8be1f8b22988a9d7b5c1b8ca6cfd56b633b0ffb2/src/Symfony/Bundle/FrameworkBundle/composer.json#L28

If this is not the right file, please tell me the path of the file, because this PR didn't changed any file in the Framework namespace and I don't know what is the relevant file.

**Comment by GromNaN** [comment]
My bad. That's the workflow component that needs this update. Sorry for the confusion.
https://github.com/symfony/symfony/blob/4.4/src/Symfony/Component/Workflow/composer.json

**Comment by fancyweb** [comment]
@alexislefebvre I am very sorry but @nicolas-grekas warned me that such changes are not made on the lowest maintained branch anymore (4.4) to avoid as many possible regressions as possible. The policy has changed recently and I did not know about it. It was not the case yet for https://github.com/symfony/symfony/pull/41576, that's why this one was made on 4.4.

However it does not mean your work is lost, you just have to rebase it on 6.1, I guess most occurrences should still be there :-) @nicolas-grekas Can you confirm it plz? :-)

**Comment by nicolas-grekas** [comment]
I do confirm yes, we decided to be stricter now and merge only true bug fixes on lower branches.

**Comment by alexislefebvre** [comment]
Thanks for your feedbacks. Psalm is trying to merge this PR in 4.4 and it fails. I opened a new PR: #44499

**Pull request closed without merging**